src/nacl/bindings/crypto_pwhash_scryptsalsa208sha256.py

```diff
@@ -0,0 +1,134 @@
+                                       ):
+    """
+    returns uses the ``passwd`` and ``salt`` to produce derive a key
+    of ``outlen`` bytes can be tuned by picking different
+    ``opslimit`` and ``memlimit``.
+
+    The constants
+        - :py:const:`.OPSLIMIT_INTERACTIVE`
+        - :py:const:`.MEMLIMIT_INTERACTIVE`
+        - :py:const:`.OPSLIMIT_SENSITIVE`
+        - :py:const:`.MEMLIMIT_SENSITIVE`
+
+    are provided as a guidance for correct settings respectively for the
+    interactive login and the long term key protecting sensitive data
+    usage cases.
+
+    :param outlen: int
+    :param passwd: bytes
+    :param salt: bytes
+    :param opslimit: int
+    :param memlimit: int
+    :rtype: bytes
+    """
+
+    if len(salt) != SALTBYTES:
+        raise ValueError("Invalid salt")
+
+    buf = ffi.new("unsigned char[]", outlen)
+
+    ret = lib.crypto_pwhash_scryptsalsa208sha256(buf, outlen, passwd,
+                                                 len(passwd), salt,
+                                                 opslimit, memlimit)
+
+    if ret != 0:
+        raise CryptoError("Key derivation fails!")
+
+    return ffi.buffer(buf, outlen)[:]
+
+
+def crypto_pwhash_scryptsalsa208sha256_str(passwd,
+                                           opslimit=OPSLIMIT_INTERACTIVE,
+                                           memlimit=MEMLIMIT_INTERACTIVE
+                                           ):
+    """
+    returns uses the ``passwd`` and ``salt`` and hashes them, producing a
+    ASCII string of STRBYTES in length, including the null terminator.
+
+    The returned string includes the salt and the tuning parameters,
+    ``opslimit`` and ``memlimit``, and can be written directly to disk
+    as a password hash
+
+    :param passwd: bytes
+    :param opslimit: int
+    :param memlimit: int
+    :rtype: bytestring
+    """
+    buf = ffi.new("unsigned char[]", STRBYTES)
+
+    ret = lib.crypto_pwhash_scryptsalsa208sha256_str(buf, passwd,
+                                                     len(passwd),
+                                                     opslimit,
+                                                     memlimit)
+
+    if(ret != 0):
+        raise CryptoError("Failed to hash password")
+
+    return ffi.string(buf)
+
+
+def crypto_pwhash_scryptsalsa208sha256_str_verify(passwd_hash, passwd):
+    """
+    Verifies the ``passwd`` against the ``passwd_hash`` that was generated.
+    Returns True or False depending on the success
+
+    :param passwd_hash: bytes
+    :param passwd: bytes
+    :rtype: boolean
+    """
+
+    if len(passwd_hash) != STRBYTES - 1:
+        raise ValueError("Invalid password hash")
+
+    if lib.crypto_pwhash_scryptsalsa208sha256_str_verify(passwd_hash,
+                                                         passwd,
+                                                         len(passwd)) == 0:
+        return True
+    else:
+        return False
```

Comment:
Inherited from owockEznan d819fbcf; while I can see the merits of both options, I've always considered a password checker a (strange) form of equality checker.
As a matter of facts, I just went to passlib's documentation at https://passlib.readthedocs.io/en/stable/ and I see on the very first page an example of an has verifier returning True/False, instead of raising an exception.